0016/README.md

```diff
@@ -0,0 +1,490 @@
+
+*Endpoint token validation response* message MUST be sent by provider in response to an [endpoint token validation request message](#endpoint-token-validation-request).
+Provider MUST publish endpoint token validation response message to the subject provided in the NATS `replyTo` field of the request.
+
+Endpoint token validation response message payload MUST be an Avro-encoded object with the following schema ([0016-endpoint-token-validation-response.avsc](./0016-endpoint-token-validation-response.avsc)):
+```json
+{
+    "namespace":"org.kaaproject.ipc.ecap.gen.v1",
+    "name":"EndpointTokenValidationResponse",
+    "type":"record",
+    "doc":"EP token validation response message",
+    "fields":[
+        {
+            "name":"correlationId",
+            "type":"string",
+            "doc":"Message ID primarily used to track message processing across services"
+        },
+        {
+            "name":"timestamp",
+            "type":"long",
+            "doc":"Message creation UNIX timestamp in milliseconds"
+        },
+        {
+            "name":"timeout",
+            "type":"long",
+            "default":0,
+            "doc":"Amount of milliseconds (since the timestamp) until the message expires. Value of 0 is reserved to indicate no expiration."
+        },
+        {
+            "name":"endpointId",
+            "type":[
+                "string",
+                "null"
+            ],
+            "doc":"Identifier of the endpoint resolved by the provided token. May be null in case token is not found."
+        },
+        {
+            "name":"statusCode",
+            "type":"int",
+            "doc":"HTTP status code of the request processing"
+        },
+        {
+            "name":"reasonPhrase",
+            "type":[
+              "null",
+              "string"
+            ],
+            "default":null,
+            "doc":"Human-readable status reason phrase"
+        }
+    ]
+}
+```
+
+
+#### Endpoint token revocation event
+
+Provider MUST publish *endpoint token revoked event* whenever one or more previously valid endpoint tokens become unusable for endpoint identification.
+On receipt of this event consumers MUST take immediate measures for terminating active endpoint communication sessions where listed tokens are used for endpoint identification.
+
+Endpoint token revoked event is a [broadcast message](/0003/README.md#broadcast-messaging) with `endpoint` target entity type, `token` event group, and `revoked` event type.
+
+Providers MUST use the following NATS subject format for endpoint token revoked events:
+```
+kaa.v1.events.{originator-service-instance-name}.endpoint.token.revoked
+```
+
+Endpoint token revoked event message payload MUST be an Avro-encoded object with the following schema ([0016-endpoint-token-revoked.avsc](./0016-endpoint-token-revoked.avsc)):
+
+```json
+{
+    "namespace":"org.kaaproject.ipc.ecap.gen.v1",
+    "name":"EndpointTokenRevokedEvent",
```

Comment:
We have `Suspended` and `Revoked` status. This event will be published on a transition to both these statuses?

Reply:
Yes, since in both of these states the token is not usable for EP identification. This is all that matters for communications service.